lib/routers/profile.js

```diff
@@ -1,7 +1,50 @@
 const { Router } = require('express');
+const isUUID = require('uuid-validate');
+const { NotFoundError } = require('../errors');
+const permissions = require('../middleware/permissions');
 
 const router = Router({ mergeParams: true });
 
+const submit = (action) => {
+  return (req, res, next) => {
+    const params = {
+      action,
+      model: 'invitation',
+      data: { ...req.body, establishment: req.establishment.id },
+      id: res.profile && res.profile.id
+    };
+    req.workflow(params)
+      .then(response => {
+        res.response = response;
+        next();
+      })
+      .catch(next);
+  };
+};
```

Comment:
We'll likely want to abstract some of this at some point, but that point needn't be now.

Reply:
tru dat